packages/javascript/bh-shared-ui/src/components/FileUploadDialog/FileUploadDialog.tsx

```diff
@@ -47,77 +45,74 @@ const FileUploadDialog: React.FC<{
         handleSubmit,
         handleRemoveFile,
         onClose,
-    } = useFileUploadDialogHandlers({ onCloseProp, dialogRef });
+    } = useFileUploadDialogHandlers({ onCloseProp });
 
     return (
         <Dialog
             open={open}
             fullWidth={true}
             maxWidth={'sm'}
+            scroll='paper'
+            onClose={onClose}
             TransitionProps={{
                 onExited: () => {
                     setFileUploadStep(FileUploadStep.ADD_FILES);
                     setFilesForIngest([]);
                 },
             }}>
-            <div ref={dialogRef}>
-                <DialogContent>
-                    <div className='pb-2 font-bold'>{headerText}</div>
-                    {description && <div>{description}</div>}
-                    <>
-                        <FileDrop
-                            onDrop={handleFileDrop}
-                            disabled={currentlyUploading || getFileUploadAcceptedTypes.isLoading}
-                            accept={getFileUploadAcceptedTypes.data?.data ?? []}
-                        />
-                        {uploadMessage && <Box className='mt-2 mb-2'>{uploadMessage}</Box>}
-                        <Link to='/administration/file-ingest' onClick={onClose}>
-                            <div className='text-center m-2 p-2 hover:bg-slate-200 rounded-md'>
-                                View File Ingest History
-                            </div>
-                        </Link>
+            <DialogTitle>
+                <div className='pb-2 font-bold'>{headerText}</div>
+                {description && <div>{description}</div>}
+
+                <FileDrop
+                    onDrop={handleFileDrop}
+                    disabled={currentlyUploading || getFileUploadAcceptedTypes.isLoading}
+                    accept={getFileUploadAcceptedTypes.data?.data ?? []}
+                />
+                {uploadMessage && <div className='mt-2 mb-2 font-normal'>{uploadMessage}</div>}
+                <Link to='/administration/file-ingest' onClick={onClose}>
+                    <div className='text-center font-normal m-2 p-2 hover:bg-slate-200 rounded-md'>
+                        View File Ingest History
+                    </div>
+                </Link>
```

Comment:
Mind swapping this Link component for the AppLink component? It supports passing along "global" query params like environmentId (which is only applicable to BHE at the moment).